Userland/Libraries/LibWeb/Layout/LineBox.cpp

```diff
@@ -33,27 +33,38 @@ void LineBox::add_fragment(Node const& layout_node, int start, int length, CSSPi
 
 void LineBox::trim_trailing_whitespace()
 {
-    while (!m_fragments.is_empty() && m_fragments.last().is_justifiable_whitespace()) {
-        auto fragment = m_fragments.take_last();
-        m_width -= fragment.width();
-    }
+    auto should_trim = [](LineBoxFragment* fragment) {
+        auto ws = fragment->layout_node().computed_values().white_space();
+        return ws == CSS::WhiteSpace::Normal || ws == CSS::WhiteSpace::Nowrap || ws == CSS::WhiteSpace::PreLine;
+    };
 
-    if (m_fragments.is_empty())
-        return;
+    LineBoxFragment* last_fragment;
+    for (;;) {
+        if (m_fragments.is_empty())
+            return;
+        last_fragment = &m_fragments.last();
+        if (!should_trim(last_fragment))
+            return;
+        if (last_fragment->is_justifiable_whitespace()) {
+            m_width -= last_fragment->width();
+            m_fragments.remove(m_fragments.size() - 1);
+        } else {
+            break;
+        }
+    }
 
-    auto& last_fragment = m_fragments.last();
-    auto last_text = last_fragment.text();
+    auto last_text = last_fragment->text();
```

Comment:
What happens if `last_fragment` is still null here?

Reply:
It cannot be null, we return on m_fragments.is_empty(). I added a comment to that effect in the code now.

I can also change it back to using a reference, it just felt this way was a bit less awkward to express, to not put is_empty() branches multiple times in the code. But if it's bad practice / more error prone I can rewrite with reference :)